packages/features/calendars/DatePicker.tsx

```diff
@@ -144,6 +146,32 @@ const Days = ({
     days.push(date);
   }
 
+  const daysToRender = days.map((day) => {
+    if (!day) return { day: null, disabled: true };
+    return {
+      day: day,
+      disabled:
+        (includedDates && !includedDates.includes(yyyymmdd(day))) || excludedDates.includes(yyyymmdd(day)),
+    };
+  });
+
+  const firstAvailableDate = daysToRender.find((day) => !day.disabled)?.day;
+
+  const doesMonthContainAvailableDates = firstAvailableDate ? true : false;
+
+  const isSelectedDateAvailable =
+    selected && doesMonthContainAvailableDates
+      ? daysToRender.some(({ day, disabled }) => {
+          if (day && yyyymmdd(day) === yyyymmdd(selected) && !disabled) return true;
+        })
+      : false;
+
+  useEffect(() => {
```

Comment:
We don't need to use effect here
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
Made some changes that avoid it. When I avoided that, a problem that I saw was that includedDates was initially empty array and later on it get's populated which was causing disabled logic to initially show all dates as enabled and then later some of the dates got disabled.
I fixed that by making sure `includedDates` is null when slots are loading and after that it becomes an array. This was anyway misleading as empty array(earlier) could mean slots loading or no slots.